File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/presentation/ui/base/activity/ActivityPresenter.java
package com.lattis.ellipse.presentation.ui.base.activity;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.lattis.ellipse.Utils.FirebaseUtil;
import com.lattis.ellipse.presentation.ui.base.BasePresenter;
import com.lattis.ellipse.presentation.ui.base.BaseView;

public abstract class ActivityPresenter<View extends BaseView> extends BasePresenter<View> {

    void onCreate(@Nullable Bundle arguments, View view) {
        setView(view);
        setup(arguments);
        setSetupState(true);
    }

    void onReCreate(@NonNull Bundle savedInstanceState, View view){
        setView(view);
        if(!isSetup()){
            setup(savedInstanceState);
            setSetupState(true);
        }
    }

    void onDestroy(boolean isFinishing) {
        clearView();

        if (isFinishing) {
            finish();
        } else {
            onTempDestroy();
        }
    }

    protected void onTempDestroy() { }

    protected void onReenter() { }

    protected void logCustomException(Throwable e){
        FirebaseUtil.getInstance().logException(e);
    }

}

File: velolabs_repos/operator-android/app/src/main/java/io/lattis/operator/presentation/base/activity/ActivityPresenter.kt
package io.lattis.operator.presentation.base.activity

import android.os.Bundle
import io.lattis.operator.presentation.base.BasePresenter
import io.lattis.operator.presentation.base.BaseView

abstract class ActivityPresenter<View : BaseView> : BasePresenter<View>() {
    var currentPhotoPath: String?=null
    fun onCreate(arguments: Bundle?, view: View) {
        setView(view)
        setup(arguments)
        setSetupState(true)
    }

    fun onReCreate(savedInstanceState: Bundle, view: View) {
        setView(view)
        if (!isSetup) {
            setup(savedInstanceState)
            setSetupState(true)
        }
    }

    fun onDestroy(isFinishing: Boolean) {
        clearView()
        if (isFinishing) {
            finish()
        } else {
            onTempDestroy()
        }
    }

    protected fun onTempDestroy() {}
    fun onReenter() {}
}